sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.c

```diff
@@ -123,9 +123,10 @@ static inline _nib_onl_entry_t *_cache_out_onl_entry(const ipv6_addr_t *addr,
             res = tmp;
             res->mode = _EMPTY;
             _override_node(addr, iface, res);
-            /* cstate masked above already */
-            res->info = cstate;
+            /* cstate masked in _nib_nc_add() already */
+            res->info |= cstate;
             res->mode = _NC;
+            clist_rpush(&_next_removable, (clist_node_t *)res);
```

Comment:
Huh? I'm traversing the FIFO here by poping and pushing the entry back in (if not gc-able), so I can't put the pop in the if-block, because if I pop or not has nothing to do with the fact if the entry is gc-able. Regarding the push, I can try.